lib/Sema/TypeCheckPattern.cpp

```diff
@@ -738,9 +738,14 @@ static bool validateParameterType(ParamDecl *decl, DeclContext *DC,
                                   TypeChecker &TC) {
   if (auto ty = decl->getTypeLoc().getType())
     return ty->is<ErrorType>();
-  
+
+  // If the element is a variadic parameter, resolve the parameter type as if
+  // it were in non-parameter position, since we want functions to be
+  // @escaping in this case.
```

Comment:
This monstrosity compiles (but shouldn't):

```swift
func foo(_ fns: @autoclosure () -> Int...) {
  fns.forEach { (x: () -> Int) in _ = print(x()) }
}

foo(1 + 2, 2 + 3, 3 + 4)
```

But I pretty much always have to use an explicit type somewhere, or it gets confused.

In retrospect, I think it was just variadic nonescaping closures we were talking about:

```swift
func foo(_ fns: () -> Int...) {
  for fn in fns {
    print(fn())
  }
}

foo({ 1 + 2 }, { 2 + 3 }, { 3 + 4 })
```